### DISCUSSION TITLE
What am i doing wrong?

### DISCUSSION THREAD

**Discussion by MaybeDaveyy** [Kaplas80/TF3#43] [Q&A] [answered]
Hello, I'm trying to use the app, and I learned how to use most of it; thing is, I'm stuck on the last step, which is when you rebuild the assets, I followed the instructions, but I end up getting a **data** folder with the files: db.par and ui.par inside. When I try to apply these files to the game, they just end up being the regular files and the translation won't work.

There was another dude who made a discussion and had the exact same issue, and apparently, they fixed it, but they didn't say how.

**Comment by MaybeDaveyy** [comment]
Talking about Kiwami 2 btw

**Comment by Kaplas80** [comment]
Hello, I haven't been able to reproduce the error. Can you post the exact commands you are using to extract and rebuild?

**Reply by MaybeDaveyy** [reply]
cd C:\Users\David\Documents\TranslationFramework3

./TF3.CommandLine.exe extract --script kiwami2pc --install-dir C:\"Program Files (x86)"\Steam\steamapps\common\"Yakuza Kiwami 2" --output-dir C:\Users\David\Desktop\yk2files
![extractedfiles](https://user-images.githubusercontent.com/113808740/198296167-79858933-99c2-493a-9281-4cc4bd66c023.png)


./TF3.CommandLine.exe rebuild --script kiwami2pc --install-dir C:\"Program Files (x86)"\Steam\steamapps\common\"Yakuza Kiwami 2" --translation-dir C:\Users\David\Desktop\yk2t\translation --output-dir C:\Users\David\Desktop\yk2t\output
![scs](https://user-images.githubusercontent.com/113808740/198296184-0f95d0a7-c851-4bf5-8717-a3083884fda5.png)

**Reply by Kaplas80** [reply]
Ok, two more questions:

- Does the app show any warning or error when running the `rebuild` command?
- Does the `translation` directory keep the same structure (`font`, `images`, `text`) and file names of the `yk2files` directory?

And one more thing... if you want the patches to be applied, the `patches` directory must be inside the `translation` directory:

```
yk2t
  ├ output
  └ translation
      ├ font
      ├ images
      ├ patches
      └ text
```

**Reply by MaybeDaveyy** [reply]
Hello. Sorry for my late response.

The translated files have the same names as the original ones, and I forgot to make the translation folder have the same structure as the extracted files folder, so I fixed that. 
![folder1](https://user-images.githubusercontent.com/113808740/198382907-51a4f314-4d87-47bc-b4a7-32129dca2ce4.png)
![folder11](https://user-images.githubusercontent.com/113808740/198382914-74765db8-2749-4374-b0a7-9b1feb653f88.png)


But now I have encountered with another issue. The app gives me the next error message when I try to rebuild:

Unhandled exception. System.NotSupportedException: Stream does not support writing.
   at System.IO.Strategies.BufferedFileStreamStrategy.WriteByteSlow(Byte value)
   at System.IO.Strategies.BufferedFileStreamStrategy.WriteByte(Byte value)
   at System.IO.FileStream.WriteByte(Byte value)
   at Yarhl.IO.StreamFormat.StreamWrapper.WriteByte(Byte value)
   at Yarhl.IO.StreamFormat.LazyFileStream.WriteByte(Byte value)
   at Yarhl.IO.DataStream.WriteByte(Byte value)
   at TF3.Core.Converters.BinaryPatch.Apply.Convert(BinaryFormat source) in /_/src/Libraries/TF3.Core/Converters/BinaryPatch/Apply.cs:line 74
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at Yarhl.FileFormat.ConvertFormat.With(IConverter converter, Object src)
   at System.Dynamic.UpdateDelegates.UpdateAndExecute3[T0,T1,T2,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2)
   at Yarhl.FileFormat.ConvertFormat.With[TConv,TParam](TParam param, Object src)
   at Yarhl.FileSystem.Node.TransformWith[TConv,TParam](TParam param)
   at TF3.Core.GameScript.ApplyPatch(PatchInfo patchInfo, Dictionary`2 containers, String translationPath) in /_/src/Libraries/TF3.Core/GameScript.cs:line 454
   at TF3.Core.GameScript.Rebuild(String gamePath, String translationPath, String outputPath) in /_/src/Libraries/TF3.Core/GameScript.cs:line 217
   at TF3.CommandLine.Program.Rebuild(RebuildOptions options) in /_/src/Apps/TF3.CommandLine/Program.cs:line 184
   at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
   at TF3.CommandLine.Program.Main(String[] args) in /_/src/Apps/TF3.CommandLine/Program.cs:line 41
PS C:\Users\David\Documents\TranslationFramework3>

**Reply by Kaplas80** [reply]
It seems that the error happens when it tries to patch the exe file. The app makes a copy of the file in memory, but maybe it keeps the original file permissions. I'm thinking in 3 possible solutions:

- Check if `YakuzaKiwami2.exe` is read only. If so, make it writable.
- Copy the game to another folder, and use it as `--install-dir`. By default, Windows "protects" the files in `Program Files`, and it only allows administrators to make changes.
- Run TF3 as administrator.

**Reply by MaybeDaveyy** [reply]
I tried all of that, but none of them worked. :( I think it's a problem of my system.
Thank you for taking your time to trying to help me.

**Reply by MaybeDavey** [reply]
Ok so today I decided to try again, and this time worked for some reason lol

**Comment by MaybeDaveyy** [comment, marked as answer]
So, how I fixed this thing was quite simple, actually.

I just had to create 3 separated folders:
- 1 for Extraction, 1 for Translation, and another one for Rebuilding; I had all these three folders inside one main folder (but you can keep them separated if you want to).

The next step is how I solved my issue:
- After I extracted the game files in the Extraction folder, at the end of the process it created another three folders inside, which are called: font, images, and text; the only thing I had to do was copying those three folders into the Translation folder (the one in which you need to make the "patches" folder). Then I translated some text files in that folder, I used the rebuild feature, and then it just worked.
-
I also recommend that you delete the files inside the "images" folder that is inside the Translation folder (do not delete the ones on the Extraction folder), so it speeds up the rebuild process, because the app rebuilds everything inside the Translation folder (So the best way is editing the images inside the "images" folder that is inside the Extraction folder and then copying those edited images into the "images" folder inside the Translation folder).

PS: Sorry for not marking this as Answered for 3 months straight (I lost this account and I recovered it just now).
